aip/general/0236.md

```diff
@@ -141,9 +146,9 @@ message PolicyPreviewMetadata {
   absent.
   - It is present on the experiment once the "preview" method is used.
 - All `preview_metadata` fields **must** be output only.
-- `state` changes to and from `ACTIVE` and `INACTIVE` when the experiment is
-  started or stopped, which can only be done by the "preview" and "stop" custom
-  methods.
+- `state` changes between `ACTIVE` and `INACTIVE` when the experiment is
```

Comment:
is there a transitory state  that we should be using? https://google.aip.dev/216#common-states.

I'd also note that Active / Inactive seem to imply some sort of degraded state (e.g. repairing is the transitionary state). I filed #1083 to clarify the use cases of these states a bit better, but ACTIVE / SUSPENDED might be better guidance here.

Reply:
INACTIVE isn't even a common state, actually. So I'd say ideally that we use a common state. if there isn't one that's a good fit, then we should add one.

Reply:
Currently the idea is to do the following:

1. An Experiment is created -> PreviewMetadata is empty (and state is empty)
2. Start -> PreviewMetadata is created, state = ACTIVE
3. Stop -> state = INACTIVE

More than happy to replace 3 with SUSPENDED. But for step 1 we want to keep PreviewMetadata empty instead of adding a state like ACCEPTED because a list w/ filter on PolicyExperiment helps us see which Experiments are in draft vs preview

Reply:
What does an empty state mean? generally "empty" corresponds to an enum unspecified, which is the default value if the client does not recognize the state.

This means that the empty value would serve two purposes: some indicator of the initial state of the resource, and a value if the client doesn't know the new enum value. Clients may misinterpret initial state to mean something else and the user may make incorrect assumptions in that case.

Reply:
So the enum is inside a message, and I'm saying the message itself is empty.
Experiment {
  PolicyPreviewMetadata {
    enum state = 1;
  }
  PolicyPreviewMetadata metadata = 1;
}

Essentially I'm saying that the metadata message is empty. Is there still a value assigned to the enum inside the empty message? If so then I agree we should find a state, but I didn't think this was the case?

Reply:
> Essentially I'm saying that the metadata message is empty. Is there still a value assigned to the enum inside the empty message? Is this the 0 value? If so then I agree we should find a state, but I didn't think this was the case....I thought it was just empty?

messages are converted to nil pointers in languages, so yes in that case it will be empty. 

But if someone sets a different field in the message, then the state enum will be set with the default value. I understand you're saying that it's unlikely for that to happen... but to be defensive of that case is precisely why we say the first value should be unspecified: https://google.aip.dev/126#guidance.